PythonVisualizations/SimpleQueue.py

```diff
@@ -230,133 +264,127 @@ def display(self):
                 # increment xpos
                 xpos += CELL_SIZE
 
-        window.update()
+        self.rearArrow = self.createIndex(self.rear, "Rear", high=False)
+        self.frontArrow = self.createIndex(self.front, "Front", high=True)
+        self.window.update()
 
     #disable insert if queue if full, disable delete if empty
     #enable everything else without overriding queue/deque functionality
     def onOffButtons(self):
         #if it's a deque
-        if buttons[5]['relief'] == 'sunken':
-            enableButtons()
+        if self.buttons[5]['relief'] == 'sunken':
+            pass
+          #  enableButtons()
             
         #if it's a queue
         else:
-            buttons[0].config(state = NORMAL)
-            buttons[3].config(state = NORMAL)
+            self.buttons[0].config(state = NORMAL)
+            self.buttons[3].config(state = NORMAL)
         
         #in all cases: disable buttons as necessary
         if self.nItems == self.size:
-            buttons[0].config(state = DISABLED)
-            buttons[1].config(state = DISABLED)
+            self.buttons[0].config(state = DISABLED)
+            self.buttons[1].config(state = DISABLED)
             
         if self.nItems == 0:
-            buttons[2].config(state = DISABLED)
-            buttons[3].config(state = DISABLED)
+            self.buttons[2].config(state = DISABLED)
+            self.buttons[3].config(state = DISABLED)
 
-def pause(pauseButton):
-    global waitVar
-    waitVar.set(True)
 
-    pauseButton['text'] = "Play"
-    pauseButton['command'] = lambda: onClick(play, pauseButton)
 
-    canvas.wait_variable(waitVar)
-
-def play(pauseButton):
-    global waitVar
-    waitVar.set(False)
-
-    pauseButton['text'] = 'Pause'
-    pauseButton['command'] = lambda: onClick(pause, pauseButton)
-
-def onClick(command, parameter = None):
-    cleanUp()
-    #disableButtons()
-    if parameter:
-        command(parameter)
-    else:
-        command()
-    #if command not in [pause, play]:
-     #   enableButtons()
-
-def cleanUp():
-    global cleanup
-    if len(cleanup) > 0:
-        for o in cleanup:
-            canvas.delete(o)
-    outputText.set('')
-    window.update()
+    def onClick(self,command, parameter = None):
+        self.cleanUp()
+        #disableButtons()
+        if parameter:
+            command(parameter)
+        else:
+            command()
+        #if command not in [pause, play]:
+         #   enableButtons()
```

Comment:
The `VisualizationApp` class takes care of calling `cleanUp` before running operations and disabling the other operations buttons.  So, I don't think this `onClick` method is providing any benefit to SimpleQueue.py